module/parser/src/commonMain/kotlin/org/cru/godtools/shared/tool/parser/model/Tabs.kt

```diff
@@ -41,8 +41,14 @@ class Tabs : Content {
     }
 
     @RestrictTo(RestrictTo.Scope.TESTS)
-    internal constructor(parent: Base = Manifest()) : super(parent) {
-        tabs = emptyList()
+    @JsName("createTestTabs")
+    constructor(
+        parent: Base = Manifest(),
+        tabs: List<Tabs.Tab> = emptyList<Tabs.Tab>(),
+        invisibleIf: String? = null,
+        goneIf: String? = null,
+    ) : super(parent, invisibleIf = invisibleIf, goneIf = goneIf) {
+        this.tabs = tabs
     }
```

Comment:
you’ve probably noticed that several places where I’ve defined test constructors I include nested content using a function instead of just passing a flat list as the parameter. The reason I do this is because the tool model has links to parent nodes from children. Just passing in a list means that the children elements don’t actually have a reference to their parents, which would have an impact on inheriting parent styles, functionality of forms, loading resources, and potentially other issues.

I haven’t commented on it before since it hasn't had any practical effect on any tests you've written using the test methods.

```suggestion
    constructor(
        parent: Base = Manifest(),
        invisibleIf: String? = null,
        goneIf: String? = null,
        tabs: ((Tabs) -> List<Tabs.Tab>)? = null,
    ) : super(parent, invisibleIf = invisibleIf, goneIf = goneIf) {
        this.tabs = tabs?.invoke(this).orEmpty()
    }
```

usage would look like:
```
val model = Tabs {
    listOf(
        Tab(it),
        Tab(it),
    )
}
```

Reply:
also, putting the function as the last parameter provides the nice looking block DSL

Reply:
Ahh yes, I was wondering about that. Will update